### PR TITLE
Feature/max poll alternatives number

### PR DESCRIPTION
**Pull request by iorveth**
Closes https://github.com/Joystream/joystream/issues/1355
Closes https://github.com/Joystream/joystream/issues/1314
Closes https://github.com/Joystream/joystream/issues/1758
Refactors `create_thread`, as suggested in https://github.com/Joystream/joystream/issues/1346
Based on https://github.com/Joystream/joystream/pull/1753

### CODE REVIEW

**Comment by bedeho** [comment]
This has a merge conflict.

**Comment by iorveth** [comment]
Fixed

File: runtime-modules/forum/src/lib.rs [hunk 1]
```diff
@@ -1038,21 +1034,32 @@ decl_module! {
         }
 
         /// Edit post text
+        // TODO need a safer approach for frame_system call
```
Comment (bedeho) [inline runtime-modules/forum/src/lib.rs]: Why was this comment added here? I do not understand it, also the heading is "EDit post text" , which does not match the extrinsic.

**Review by bedeho — CHANGES_REQUESTED** [review]

File: runtime-modules/forum/src/lib.rs (hunk 1 above)
Comment (iorveth) [reply runtime-modules/forum/src/lib.rs]: Just moved it from the inner function, performing actual mutation. Seems, that that naming mismatch was here before, i can fix.

**Review by bedeho — APPROVED** [review]

**Pull request merged**